src/net.cpp

```diff
@@ -130,6 +130,16 @@ std::thread open_added_connections_thread;
 std::thread open_connections_thread;
 std::thread message_handler_thread;
 
+static void InterruptibleSleep(uint64_t n)
+{
+    bool ret = false;
+    {
+        std::unique_lock<std::mutex> lock(cs_net_interrupt);
+        ret = net_interrupt_cond.wait_for(lock, std::chrono::milliseconds(n), []()->bool{return net_interrupted; });
+    }
+    interruption_point(ret);
```

Comment:
might be better (although less clean) to use a noexcept version that returns a bool.